django/db/backends/sqlite3/operations.py

```diff
@@ -201,13 +201,28 @@ def sql_flush(self, style, tables, *, reset_sequences=False, allow_cascade=False
             # Simulate TRUNCATE CASCADE by recursively collecting the tables
             # referencing the tables to be flushed.
             tables = set(chain.from_iterable(self._references_graph(table) for table in tables))
-        # Note: No requirement for reset of auto-incremented indices (cf. other
-        # sql_flush() implementations). Just return SQL at this point
-        return ['%s %s %s;' % (
+        sql = ['%s %s %s;' % (
             style.SQL_KEYWORD('DELETE'),
             style.SQL_KEYWORD('FROM'),
             style.SQL_FIELD(self.quote_name(table))
         ) for table in tables]
+        if reset_sequences:
+            sequences = [{'table': table} for table in tables]
+            sql.extend(self.sequence_reset_by_name_sql(style, sequences))
+        return sql
+
+    def sequence_reset_by_name_sql(self, style, sequences):
+        return [
+            "%s %s %s %s = 0 %s %s = '%s';" % (
+                style.SQL_KEYWORD('UPDATE'),
+                style.SQL_TABLE(self.quote_name('sqlite_sequence')),
+                style.SQL_KEYWORD('SET'),
+                style.SQL_FIELD('seq'),
+                style.SQL_KEYWORD('WHERE'),
+                style.SQL_FIELD('name'),
+                sequence_info['table'],
+            ) for sequence_info in sequences
+        ]
```

Comment:
Any thoughts about issuing a single `UPDATE` statement by doing `name IN` instead?

Reply:
Good suggestion. I've implemented this in the latest revision.